src/container.ts

```diff
@@ -1,23 +1,22 @@
-import * as util from 'util'
-import { GlobWithOptions, listModules } from './list-modules'
+import { inspectCustom, req } from './deps.deno.ts'
```

Comment:
This won't break the browser builds right?

Reply:
With the current setup, the browser builds are going be derived from the build output. The build output is equivalent code to what is used for Node right now.

That being said, I am not quite sure how to change the rollup config to actually work based on JS instead of TS. For example, the bundling is done by `tsc`, correct? Which bundler could be used instead? Just let `tsc` operate on the JS?

Reply:
I'm not sure how Rollup uses the TypeScript plugin, whether its for checking and transpiling or whether it also does bundling? I would imagine that's what Rollup itself is doing?

Reply:
I'll just try to make it operate on the JS and then we can see how well this works.

Reply:
This is not a good idea because it would first emit code for Node including the module resolution things that Node needs, and then it would go on transforming this output for the browser.

It would probably be best to just use the deno tooling to create bundles. However, this would mean that we have to add a custom build script that only runs on deno. I originally advocated for option 2 in #281 because I thought it would be possible to switch this repo to deno without breaking any of the existing config. Apart from the browser builds, this was correct. However, I currently do not see how we could go on from here (apart from creating a rollup plugin for deno2node, which clearly isn't feasible).

I will think some more about this, but right now I'm not too optimistic that this undertaking will work out the way I had hoped for.

The next attempt could be to try option 3, hence fully transforming this repo into a true Deno project. From a tooling and maintenance point of view, this would be the easiest option by far, however, it also means that I'd have to completely disrupt this setup. Things like npm installs from GitHub might not work after that, or maybe they will, no idea. I do not mind spending the time on such an attempt, but since you said you had not used Deno before, I am unsure how comfortable you'd be with such a change.

Most people who try Deno love it, but I really did not come here to push anyone to change their stack, all I wanted was to add compatibility for my own needs. In other words, I understand if you are not open for a PR that changes this whole setup. In that case, I'd probably fork and roll my own DI container based on awilix.

Reply:
I'd hate for you to do all of that work and then in the end not merge the PR, but I am interested in seeing what option 3 looks like, as long as the browser builds and the Node builds will work the same as they have been (in other words, we can do this as a patch release from a semantic versioning standpoint) and if the Deno tooling works well then I wouldn't mind trying it out.

It sounds like you would be going with option 3 either way in case you fork, so I suppose it wouldn't be wasted anyway?

Reply:
I don't mind trying it out. We're talking a day of work here, not two weeks, so even it does not lead to a merge, it's fine :)

I would like it best if we would end up with a solution that works as a patch release in the existing project, and supports Deno. Starting my own project is something I'd much rather avoid. Awilix is the best of its kind, maintaining a fork does not make much sense. Hence, if this won't be merged, I would probably limit my fork to Deno so that there is no unnecessary competition on Node and in the browser.

I'll start the new attempt on a new branch.

Reply:
Going to postpone this by a few weeks. Other, more important tasks need to be prioritised. I'll get back to you once I have something.

Reply:
I'm still waiting S2!

Reply:
I currently still have other things to take care of, but this will remain on my list until it's complete—be it by me, or by anyone else who wants to join in.

There is a working version in my fork that you can use from GitHub, but it's a bit outdated by now, and it also doesn't have everything that would be needed to merge this.